### PR TITLE
Do not output KV cache when not using `with-past` in the ONNX export

### PR DESCRIPTION
**Pull request by fxmarty**
As per title, likely fixes a longstanding bug. Should anyway refactor all that to use `variant` at some point.

Fixes https://github.com/huggingface/optimum/issues/1334

### CODE REVIEW

File: optimum/exporters/onnx/base.py [hunk 1]
```diff
@@ -821,24 +821,24 @@ def __init__(
...
+            if use_present_in_outputs is None:
+                self.use_present_in_outputs = True
 
     def with_behavior(
         self,
         behavior: Union[str, ConfigBehavior],
         int_dtype: str = "int64",
         float_dtype: str = "fp32",
         use_past: bool = False,
+        use_present_in_outputs: Optional[bool] = None,
```
[the remark this answers is not in the document]
Comment (fxmarty) [reply optimum/exporters/onnx/base.py:841]: I was just unwilling to change the signature of the `__init__` that is for some reason `use_present_in_outputs: Optional[bool] = None` and to have to fix potential things breaking due to that.

**Comment by fxmarty** [comment]
Actually it happens that `use_present_in_outputs` is likely useless thanks to @echarlaix notice.

File: optimum/exporters/onnx/base.py [hunk 2]
```diff
@@ -821,24 +821,24 @@ def __init__(
...
+        # Override this to specify custom attribute change for a given behavior.
         if self._behavior is ConfigBehavior.ENCODER:
             self.task = "feature-extraction"
             self.use_past_in_inputs = False
             self.use_present_in_outputs = False
         if self._behavior is ConfigBehavior.DECODER:
             self.use_past_in_inputs = self.use_past
-            self.use_present_in_outputs = True
+            if use_present_in_outputs is None:
+                self.use_present_in_outputs = True
```
Comment (echarlaix) [inline optimum/exporters/onnx/base.py:833]: Could this be removed as well (to keep [`use_past`](https://github.com/huggingface/optimum/blob/v1.12.0/optimum/exporters/onnx/base.py#L503) default), also not sure to understand the point of `use_present_in_outputs`, when should it be different from `use_past` ? cc @michaelbenayoun worked on https://github.com/huggingface/optimum/commit/2a4b790be4ac1640e01f3f54b1381e4b594d1695

**Review by echarlaix — APPROVED** [review]

File: optimum/exporters/onnx/base.py (hunk 2 above)
Comment (fxmarty) [reply optimum/exporters/onnx/base.py:833]: As discussed offline with Michael we can remove it.

**Review by mht-sharma — APPROVED** [review]

**Comment by fxmarty** [comment]
In the end this PR removes the `use_present_in_outputs`, and changes the behavior of `use_past` which is now true for the export of both `decoder_model.onnx` and `decoder_with_past_model.onnx` when we export with `use_cache=True`. WDYT?

**Review by mgoin — APPROVED** [review]
> In the end this PR removes the use_present_in_outputs, and changes the behavior of use_past which is now true for the export of both decoder_model.onnx and decoder_with_past_model.onnx when we export with use_cache=True. WDYT?

This satisfies my use case and I've tested it works to replace the previous workaround. Appreciate it!

**Pull request merged**